x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/permission/RemoteIndicesPermission.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.core.security.authz.permission;
+
+import org.elasticsearch.common.bytes.BytesReference;
+import org.elasticsearch.core.Nullable;
+import org.elasticsearch.xpack.core.security.authz.RestrictedIndices;
+import org.elasticsearch.xpack.core.security.authz.privilege.IndexPrivilege;
+import org.elasticsearch.xpack.core.security.support.Automatons;
+import org.elasticsearch.xpack.core.security.support.StringMatcher;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+public record RemoteIndicesPermission(List<RemoteIndicesGroup> remoteIndicesGroups) {
```

Comment:
We will use this class for the pre-flight authorization check as well as to construct `RoleDescriptor.IndicesPrivileges` to send from the QC to the FC.

Reply:
Yup, common base class would make sense. This is another case where my explanatory comment got buried in hidden items: https://github.com/elastic/elasticsearch/pull/90614#discussion_r991587789

Essentially, we can have IndicesPrivileges and RemoteIndicesPrivileges inherit from a base class but that requires a bigger refactor to IndicesPrivileges which I'd like to do separately since this is already a v big PR.

Reply:
Resolving based on the other comment.